Sync/StorageClient.swift

```diff
@@ -441,63 +434,63 @@ public class Sync15StorageClient {
     }
 
     // Sync storage responds with a plain timestamp to a PUT, not with a JSON body.
-    private func putResource<T>(path: String, body: JSON, ifUnmodifiedSince: Timestamp?, parser: (String) -> T?) -> Deferred<Result<StorageResponse<T>>> {
+    private func putResource<T>(path: String, body: JSON, ifUnmodifiedSince: Timestamp?, parser: (String) -> T?) -> Deferred<Maybe<StorageResponse<T>>> {
         let url = self.serverURI.URLByAppendingPathComponent(path)
         return self.putResource(url, body: body, ifUnmodifiedSince: ifUnmodifiedSince, parser: parser)
     }
 
-    private func putResource<T>(URL: NSURL, body: JSON, ifUnmodifiedSince: Timestamp?, parser: (String) -> T?) -> Deferred<Result<StorageResponse<T>>> {
-
-        let deferred = Deferred<Result<StorageResponse<T>>>(defaultQueue: self.resultQueue)
-
-        if self.checkBackoff(deferred) {
-            return deferred
-        }
-
-        let req = self.requestPUT(URL, body: body, ifUnmodifiedSince: ifUnmodifiedSince)
-        let handler = self.errorWrap(deferred) { (_, response, data, error) in
-            if let data = data as? String {
-                if let v = parser(data) {
-                    let storageResponse = StorageResponse<T>(value: v, response: response!)
-                    deferred.fill(Result(success: storageResponse))
-                } else {
-                    deferred.fill(Result(failure: RecordParseError()))
-                }
-                return
-            }
-
-            deferred.fill(Result(failure: RecordParseError()))
-        }
-
-        // Yay Swift.
-        let stringHandler = { (a, b, c: String?, d) in
-            return handler(a, b, c, d)
-        }
-
-        req.responseString(encoding: nil, completionHandler: stringHandler)
+    private func putResource<T>(URL: NSURL, body: JSON, ifUnmodifiedSince: Timestamp?, parser: (String) -> T?) -> Deferred<Maybe<StorageResponse<T>>> {
+
+        let deferred = Deferred<Maybe<StorageResponse<T>>>(defaultQueue: self.resultQueue)
+
+//        if self.checkBackoff(deferred) {
```

Comment:
@fluffyemily commented it out because it was causing her problems. Parts of Sync will be non-functional without this.